tests/test_manager.py

```diff
@@ -0,0 +1,110 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from __future__ import print_function
+from django.db.models.expressions import F
+from django.db.models.query import QuerySet
+from django.test import TestCase
+
+from django_cte import With, CTEQuerySet, CTEManager
+
+from .models import (
+    Order,
+    OrderFromLT40,
+    OrderLT40AsManager,
+    OrderCustomManagerNQuery,
+    OrderCustomManager,
+    LT40QuerySet,
+    LTManager,
+    LT25QuerySet,
+)
+
+
+class TestCTE(TestCase):
+    def test_cte_queryset_correct_defaultmanager(self):
+        self.assertEqual(type(Order._default_manager), CTEManager)
+        self.assertEqual(type(Order.objects.all()), CTEQuerySet)
+
+    def test_cte_queryset_correct_from_queryset(self):
+        self.assertEqual(type(OrderFromLT40.objects.all()), LT40QuerySet)
+
+    def test_cte_queryset_correct_queryset_as_manager(self):
+        self.assertEqual(type(OrderLT40AsManager.objects.all()), LT40QuerySet)
+
+    def test_cte_queryset_correct_manager_n_from_queryset(self):
+        self.assertTrue(isinstance(
```

Comment:
Could use `self.assertIsInstance(...)` here (and possibly other places too). The error message is more helpful if it fails.